uplink/src/collector/tcpjson.rs

```diff
@@ -144,10 +148,26 @@ impl Bridge {
                         }
                     };
 
-                    if let Err(e) = partition.fill(data).await {
-                        error!("Failed to send data. Error = {:?}", e.to_string());
-                    }
+                    let data_stream = data.stream.clone();
+
+                    let flushed = match partition.fill(data).await {
+                        Ok(f) => f,
```

Comment:
I think we can simplify all the below `if flushed` logic into `flush_handler.reset(&stream_name)` in a match branch here

Reply:
Create a new return status `Capacity` from `fill`.

```rust
enum Capacity {
    /// First element added
    Initialized(String)
    /// Name of the stream which is full
    Flushed(String)
    /// No buffer is full
    Partial
}
```

```rust
match partitions.fill(data).await {
    Ok(Capacity::Initialized(stream)) => // Initialize timeout for this stream
    Ok(Capacity::Flushed(steam)) => // Reset timeout
    ....
}
```

Reply:
Do you still feel this is necessary?